### ISSUE TITLE
Add new builtin like nohup

### ISSUE THREAD

**Issue by Ericson2314** [fish-shell/fish-shell#4010]
In https://github.com/fish-shell/fish-shell/issues/2810i first learned of `nohup`, which actually suits my purposes better than `... & disown` due to (at least on other shells) a race if a stopped job already exists.

But nohup the binary commonly logs the standard streams to a file, and behavior I neither care for nor can turn off. It would nice to atomically fork off a process separate from the jobs list.

**Comment by krader1961** [comment]
The standard way to do that is known as the double fork trick. For example, `fish -c 'sleep 33 &'`. You can also easily and with more safety compared to other shells use a function to get the behavior you want. For example,

```
function nohup
    if isatty 1
        command nohup $argv </dev/null >/dev/null 2>&1
    else
        command nohup $argv </dev/null
    end
end
```

Unfortunately because you can't background functions doing `nohup some_cmd &` won't work. You have to press [ctrl-z] then `bg` to put the nohup'd command in the background.

It might be nice if `disown` was special like `exec` or `command` so that you could do `disown sleep 33` to launch an external command without recording it in the job table. Or perhaps `command` should have a `-d` / `--disown` flag.

**Comment by Ericson2314** [comment]
I like adding a flag to `command` the best.

**Comment by difranco** [comment]
This still doesn't seem to support the most common use case by far in my experience, which is that I start a job, it is running in the background, and I later find I want to exit my shell and leave the job running, which is what disown does in e.g. bash. This is in my opinion an essential job management function.

**Comment by zanchey** [comment]
That's because this is a different feature; you're looking for #3906 which will be in the next release.

**Comment by actionless** [comment]
this finally works as before (not exactly but could be restored with an alias), thanks a lot!

i think this one can be closed btw

**Comment by Ericson2314** [comment]
No, this shouldn't be closed because of the `disown` command. In the common case one knows they want the job disowned before it's created, so it would be nice to do that all atomically for ease of typing, avoidance of race conditions, and implementation simplicity.

**Comment by kulabun** [comment]
Actually I can't even image a situation when I may want to compound lifetime of my background tasks to my lifetime of my terminal session. But this workaround works fine:

```
function nohup
        command nohup $argv </dev/null >/dev/null 2>&1 & disown
end
```

**Comment by nhooyr** [comment]
Whats the race here that this solves?

**Comment by lynsei** [comment]
Why run a server in the foreground?  Docker setups pipe to STDOUT.  Many containers these days are single process, or multi-process a la PM2 or similar.  I'm also quite curious of the purpose of this because I can't seem to find what it is designed to prevent or stop from happening.  Any sage wisdom?
